docs/tutorial.rst

```diff
@@ -306,17 +318,281 @@ trees. In this example we incorrectly inferred that 0 coalesces with the
 ancestor of nodes 2, 3, 4 and 5 before 1.
 
 
-.. todo::
+************
+Data example
+************
+
+Inputting real data for inference is similar in principle to the examples above:
+you simply iterate over the sites in your data file, calling :meth:`.SampleData.add_site`
+for each variable site. We do not provide methods to read directly from other formats, as
+these formats are often too complex for automatic conversion. However, many Python
+libraries are available for reading different genomic data formats, and writing a script
+to iterate over the sites in such files is usually a fairly simple task. The example
+below shows how this can be done for a VCF file, using a freely available VCF reading
+library.
+
++++++++++++++
+Reading a VCF
++++++++++++++
+
+A common way to store genetic variation data is as a VCF (Variant Call Format) file.
+The `CYVCF2 library <https://github.com/brentp/cyvcf2>`_ provides a relatively fast
+Python interface for reading VCF files. The following script demonstrates how to infer a
+tree sequence from a simple VCF file: in particular the function below named
+``add_diploid_sites`` illustrates how to iterate over the variants in a ``CYVCF2.VCF``
+object and add them to a ``tsinfer`` sample data file. This particular script assumes
+that the VCF describes a set of phased, diploid individuals, and has the ancestral state
+present in the INFO field of the VCF (otherwise it takes the reference allele as the
+ancestral state). As example data, we use a publicly available VCF file of the genetic
+variants from chromosome 24 of ten Norwegian and French house sparrows,
+*Passer domesticus* (thanks to Mark Ravinet for the data file):
+
+.. code-block:: python
+
+    import cyvcf2
+    import tsinfer
+    
+    def add_diploid_sites(vcf, samples):
+        pos = None
+        for variant in vcf:
+            if pos == variant.POS:
+                raise ValueError("Duplicate positions for variant at position", pos)
+            else:
+                pos = variant.POS
+            if any([row[2]==False for row in variant.genotypes]):
+                raise ValueError("Unphased genotypes for variant at position", pos)
+            alleles = [variant.REF] + variant.ALT
+            # Sort with the ancestral allele first if we have one (otherwise use REF)
+            ancestral = variant.INFO.get('AA', variant.REF)
```

Comment:
Why not
```python
ancestral = variant.INFO.get('AA', variant.REF)
all_alleles = set(variant.ALT + [variant.REF])
# Ancestral state must be first in the allele list.
alleles = [ancestral] + list(all_alleles - {ancestral})
# Map allele values to their indexes in the alleles list for each sample.
genotypes = [alleles.index(allele) for row in variant.genotypes for allele in row[0:2]]
```
All the sorting stuff is obscuring the point of the operations, to me.

Reply:
Yes, this is nicer. Thanks.

Reply:
Although the VCF (and CYVCF2) seems to list `0`s and `1`s, not e.g. `A`s and `T`s so we don't want to be using `.index`. Rather, we want to make a list of the variants, and reorder somehow. I'll see what I can do.

Reply:
So I think this is how you need to do it using your approach:

```
            all_alleles = [variant.REF] + variant.ALT
            # Ancestral state must be first in the allele list.
            alleles = [ancestral] + list(set(all_alleles) - {ancestral})
            allele_index = {old_index: alleles.index(allele) for old_index, allele in enumerate(alleles)}
            # Map original allele indexes to their indexes in the new alleles list.
            genotypes = [allele_index[old_index] for row in variant.genotypes for old_index in row[0:2]]
```